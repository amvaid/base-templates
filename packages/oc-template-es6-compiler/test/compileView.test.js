jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000;

const path = require('path');
const fs = require('fs-extra');
const compileView = require('../lib/compileView.js');
const _ = require('lodash');

const componentPath = path.join(__dirname, '../../../mocks/es6-component');
const publishPath = path.join(componentPath, '_package');
const componentPackage = require(`${componentPath}/package.json`);
const publishFileName = 'template.js';

const options = {
  componentPackage,
  componentPath,
  publishPath,
  publishFileName
};

test('Should correctly compile the view', done => {
  compileView(options, (err, compiledViewInfo) => {
    expect(compiledViewInfo).toMatchSnapshot();
    expect(
      fs.readFileSync(path.join(publishPath, publishFileName), 'UTF8')
    ).toMatchSnapshot();
    fs.removeSync(path.join(publishPath, publishFileName));
    done();
  });
});
