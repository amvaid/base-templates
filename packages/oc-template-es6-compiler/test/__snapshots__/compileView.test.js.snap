// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the view 1`] = `
Object {
  "template": Object {
    "hashKey": "bf6f2d67586ec877da0348aef6981a75891ab353",
    "src": "template.js",
    "type": "oc-template-es6",
  },
}
`;

exports[`Should correctly compile the view 2`] = `"var oc=oc||{};oc.components=oc.components||{},oc.components.bf6f2d67586ec877da0348aef6981a75891ab353=\\"<h1>Hello {name.toUpperCase()}!</h1>\\";"`;
